client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/list-item.js

```diff
@@ -0,0 +1,166 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+import { trim } from 'lodash';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import CarrierIcon from '../../components/carrier-icon';
+import Dialog from 'components/dialog';
+import {
+	getCarrierAccountsState,
+} from 'woocommerce/woocommerce-services/state/carrier-accounts/selectors';
+import {
+	setVisibilityDisconnectCarrierDialog,
+	disconnectCarrier,
+} from 'woocommerce/woocommerce-services/state/carrier-accounts/actions';
+
+const CarrierAccountListItem = ( props ) => {
+	const {
+		isPlaceholder,
+		data,
+		children,
+		translate,
+		siteId,
+		showDisconnectDialog,
+	} = props;
+	if ( isPlaceholder ) {
+		return (
+			<div className="carrier-accounts__list-item">
+				<div className="carrier-accounts__list-item-placeholder-carrier-icon">
+					<div className="carrier-accounts__list-item-carrier-icion-placeholder">
+						<span />
+					</div>
+				</div>
+				<div className="carrier-accounts__list-item-name">
+					<div className="carrier-accounts__list-item-name-placeholder">
+						<span />
+					</div>
+				</div>
+				<div className="carrier-accounts__list-item-actions">{ children }</div>
+			</div>
+		);
+	}
+
+	const renderIcon = carrierId => {
+		return <div className="carrier-accounts__list-item-carrier-icon">
+			<CarrierIcon carrier={ carrierId } size={ 18 } />
+		</div>;
+	};
+
+	const renderName = name => {
+		const carrierName = name && '' !== trim( name ) ? name : translate( 'Untitled' );
+		return <div className="carrier-accounts__list-item-name">
+			<span>{ carrierName }</span>
+		</div>;
+	};
+
+	const renderCredentials = credentials => {
+		return <div className="carrier-accounts__list-item-credentials">
+			<span>{ credentials }</span>
+		</div>;
+	}
+
+	const showDisconnectDialogHandler = () => {
+		props.setVisibilityDisconnectCarrierDialog( siteId, data.carrierId, true );
+	}
+
+	const hideDisconnectDialogHandler = () => {
+		props.setVisibilityDisconnectCarrierDialog( siteId, data.carrierId, false );
```

Comment:
Is the show/hide for dialog a state only within `CarrierAccountListItem`? Curious why global state store (redux) is preferred over using local state?

Reply:
this is probably a personal preference due to my Elm background. The global store is actually contextualized per view/component in the plugin itself: other views like the shipping label is using the global store to persist some views' state. I've had bad experiences in the past using the global redux store + components' state, and also I think the recommended path in the past in the React community was to only use the redux store if you're already using it. But, if you think we should store this state in the component, I'm totally fine with it too!

Reply:
Ah I see. Thanks!